src/module_loader.cc

```diff
@@ -44,39 +45,63 @@
 
 namespace {
 
+// Strip the '@' prefix from command categories (e.g., @read)
+// to format them for Valkey Search's prefix ACL rules (e.g., read).
+inline std::string ACLPermissionFormatter(
+    const absl::flat_hash_set<absl::string_view> &cmd_permissions) {
+  std::vector<absl::string_view> permissions;
+  for (auto permission : cmd_permissions) {
+    CHECK(permission[0] == '@');
+    permissions.push_back(permission.substr(1));
+  }
+  return absl::StrJoin(permissions, " ");
+}
+
 vmsdk::module::Options options = {
     .name = "search",
+    .acl_categories =
+        {
+            "search",
+        },
     .version = MODULE_VERSION,
     .info = valkey_search::ModuleInfo,
     .commands =
         {
             {
                 .cmd_name = valkey_search::kCreateCommand,
-                .permissions = "write deny-oom",
+                .permissions = ACLPermissionFormatter(
+                    valkey_search::kCreateCmdPermissions),
+                .deny_oom = true,
```

Comment:
So is deny_oom pulled out of the permissions since it isn't a command category?

To me, it might make sense to have two fields, one for "categories" and one for "permissions"? This way deny_oom isn't special cased, e.g. if there are other non-category permissions we want to add later.